src/callbacks/amr.jl

```diff
@@ -0,0 +1,302 @@
+
+"""
+    AMRCallback(semi, indicator [,adaptor=AdaptorAMR(semi)];
+                interval=5,
+                adapt_initial_conditions=true,
+                adapt_initial_conditions_only_refine=true)
+
+Performs adaptive mesh refinement (AMR) every `interval` time steps
+for a given semidiscretization `semi` using the chosen `indicator`.
+"""
+struct AMRCallback{Indicator, Adaptor}
+  indicator::Indicator
+  interval::Int
+  adapt_initial_conditions::Bool
+  adapt_initial_conditions_only_refine::Bool
+  adaptor::Adaptor
+end
+
+
+function AMRCallback(semi, indicator, adaptor; interval=5,
+                                               adapt_initial_conditions=true,
+                                               adapt_initial_conditions_only_refine=true)
+  # AMR every `interval` time steps
+  condition = (u, t, integrator) -> interval > 0 && (integrator.iter % interval == 0)
```

Comment:
Good question. I used these checks to basically use the same logic as before. I don't think it's easier to add another line `active=false` when creating a specific callback than just copying and editing the line `callbacks = CallbackSet(...)`. 

It might make sense to use an `AMRCallback` with `interval=0` if one only wants to adapt the mesh to the initial condition during initialization of the callback. If we don't check whether `interval > 0`, we might get a `DivideError: integer division error`. What about changing the `condition` to
```julia
if interval > 0
  condition = (u, t, integrator) -> integrator.iter % interval == 0
else # disable the AMR callback except possibly for initial refinement during initialization
  condition = (u, t, integrator) -> false
end
```

Reply:
I didn't think about just excluding the callback from the callbacks set - this is fine by me as well.

I agree on the second paragraph - the proposed change in the `condition` sounds good to me!

Reply:
dfcdde0